docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+# Service Catalog API
+
+This document defines a Kubernetes-native service broker API. It is a formalization of [@krancour](http://github.com/krancour)'s [presentation](https://docs.google.com/presentation/d/1Fm3qG9zQ4R8ZbhtTObmQjO3UZYJ0ruYI4qr5w_gEKN8/edit?usp=sharing) that he gave to the sig-service-catalog meeting on 10/10/2016.
```

Comment:
Can we remove the "history text"?  We can move this into the PR notes if we want to keep it, but the api docs themselves should be clean of this type of stuff and just state what the API is.